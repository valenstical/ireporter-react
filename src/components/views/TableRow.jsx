import React from 'react';
import { Link } from 'react-router-dom';
import ago from 'date-fns/distance_in_words_strict';
import DateFormat from 'date-fns/format';
import PropTypes from 'prop-types';

const getStatus = {
  draft: { icon: 'file-text-o', class: 'draft' },
  resolved: { icon: 'check', class: 'resolved' },
  rejected: { icon: 'remove', class: 'rejected' },
  'under investigation': { icon: 'info-circle', class: 'investigating' },
};


/**
 * The table row component
 * @returns {object} The generated JSX object
 */
export default function TableRow({ report }) {
  const status = getStatus[report.status];
  return (
    <tr>
      <td className="td-small" title={DateFormat(report.createdOn, 'MMM Do, YYYY h:mm a')}>
        {DateFormat(report.createdOn, 'MMM Do, YYYY')}
        <br />
        <small>
(
          {`${ago(report.createdOn, new Date())} ago`}
)
        </small>
      </td>
      <td>
        <i className={`fa flag ${report.type === 'red-flag' ? 'fa-flag flag-red-flag' : 'fa-bullhorn flag-intervention'}`} title={report.type} />
        <Link to={`/reports/${report.type}s/${report.id}`}><ins>{report.title}</ins></Link>
      </td>
      <td className="relative">
        <div className="profile-img">
          <div className="profile-img-reports" style={{ backgroundImage: `url(${report.profile})` }} />
        </div>
        {' '}
        <span className="profile-name text-capitalize">{report.firstname}</span>
      </td>
      <td className="text-center">
        <span className={`badge-citizen badge-status status-badge-${status.class}`}>
          <i className={`fa fa-${status.icon}`} />
          {status.class}
        </span>
      </td>
    </tr>
  );
}

TableRow.propTypes = {
  report: PropTypes.object.isRequired,
};
