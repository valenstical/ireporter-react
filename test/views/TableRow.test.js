import React from 'react';
import { shallow } from 'enzyme';
import TableRow from '../../src/components/views/TableRow';

const report = {
  status: 'draft',
  type: 'red-flag'
};

describe('<TableRow>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<TableRow report={report} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing for intervention reports', () => {
    report.type = 'intervention';
    const wrapper = shallow(<TableRow report={report} />);
    expect(wrapper).toMatchSnapshot();
  });
});
