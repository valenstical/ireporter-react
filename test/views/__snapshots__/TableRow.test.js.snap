// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRow> should render without crashing 1`] = `
<tr>
  <td
    className="td-small"
    title="Invalid Date"
  >
    Invalid Date
    <br />
    <small>
      (
      NaN years ago
      )
    </small>
  </td>
  <td>
    <i
      className="fa flag fa-flag flag-red-flag"
      title="red-flag"
    />
    <Link
      replace={false}
      to="/reports/red-flags/undefined"
    >
      <ins />
    </Link>
  </td>
  <td
    className="relative"
  >
    <div
      className="profile-img"
    >
      <div
        className="profile-img-reports"
        style={
          Object {
            "backgroundImage": "url(undefined)",
          }
        }
      />
    </div>
     
    <span
      className="profile-name text-capitalize"
    />
  </td>
  <td
    className="text-center"
  >
    <span
      className="badge-citizen badge-status status-badge-draft"
    >
      <i
        className="fa fa-file-text-o"
      />
      draft
    </span>
  </td>
</tr>
`;

exports[`<TableRow> should render without crashing for intervention reports 1`] = `
<tr>
  <td
    className="td-small"
    title="Invalid Date"
  >
    Invalid Date
    <br />
    <small>
      (
      NaN years ago
      )
    </small>
  </td>
  <td>
    <i
      className="fa flag fa-bullhorn flag-intervention"
      title="intervention"
    />
    <Link
      replace={false}
      to="/reports/interventions/undefined"
    >
      <ins />
    </Link>
  </td>
  <td
    className="relative"
  >
    <div
      className="profile-img"
    >
      <div
        className="profile-img-reports"
        style={
          Object {
            "backgroundImage": "url(undefined)",
          }
        }
      />
    </div>
     
    <span
      className="profile-name text-capitalize"
    />
  </td>
  <td
    className="text-center"
  >
    <span
      className="badge-citizen badge-status status-badge-draft"
    >
      <i
        className="fa fa-file-text-o"
      />
      draft
    </span>
  </td>
</tr>
`;
